### PR TITLE
py-onnx: Update to version 1.8.1, Use Python39 as default

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-onnx/Portfile [hunk 1]
```diff
@@ -30,12 +30,12 @@ long_description    Open Neural Network Exchange (ONNX) is an open\
...
-checksums           rmd160  8d2fc7d52238599388298fbe7e58f85d2ffcc96b \
-                    sha256  42f2aa3982964511a880e0833d6cdc3faa8ea69158ce0701fcb0b07db4a95777 \
-                    size    4930674
+checksums           rmd160  a319bd2f15fbbd3964e0751bd6e90596efdc6c0f \
+                    sha256  08e5f1b0a3a0a7d3bc608c6d95f3cabcd22b23148a791418e64d8dcfe3e7a0a4 \
+                    size    4934008
 
 python.versions     36 37 38 39
-python.default_version 38
+python.default_version 39
```
Comment (reneeotten) [inline python/py-onnx/Portfile:38]: why not remove this line completely and let the python PG file set this?

Comment (essandess) [reply python/py-onnx/Portfile]: Done. I also removed Python 36.

Comment (reneeotten) [reply python/py-onnx/Portfile]: that sounds like a bad idea:
```
% port echo depends:py36-onnx
py36-pytorch
```

Comment (cjones051073) [reply python/py-onnx/Portfile]: I noted that. pytorch is a big build so I think dropping py36 there as well is probably a good idea at this time.

Comment (essandess) [reply python/py-onnx/Portfile]: Looks like that's been done already with an update too, thanks if it was you.

Comment (cjones051073) [reply python/py-onnx/Portfile]: yep, as it happened I was testing an update to pytorch anyway.

**Pull request merged**